client/src/components/profile/components/HeatMap.js

```diff
@@ -20,29 +22,32 @@ const propTypes = {
   })
 };
 
-const now = Date.now();
-const today = new Date(now);
-const sixMonthsInMillseconds = 15780000000;
-const sixMonthsAgoInMilliseconds = now - sixMonthsInMillseconds;
-const sixMonthsAgo = startOfMonth(sixMonthsAgoInMilliseconds);
-
 function HeatMap({ calendar, streak }) {
-  const dateValueMap = Object.keys(calendar)
-    .map(ts => Number(ts * 1000) || null)
-    .filter(Boolean)
-    .reduce((map, current) => {
-      const startOfCurrent = format(startOfDay(current), 'YYYY-MM-DD');
-      if (startOfCurrent in map) {
-        map[startOfCurrent] = map[startOfCurrent] + 1;
-      } else {
-        map[startOfCurrent] = 1;
-      }
-      return map;
-    }, {});
+  const startOfToday = startOfDay(Date.now());
+  const sixMonthsAgo = addMonths(startOfToday, -6);
+  const startOfCalendar = format(addDays(sixMonthsAgo, -1), 'YYYY-MM-DD');
+  const endOfCalendar = format(startOfToday, 'YYYY-MM-DD');
+
+  let calendarData = {};
+  let dayCounter = sixMonthsAgo;
+
+  while (dayCounter <= startOfToday) {
+    calendarData[format(dayCounter, 'YYYY-MM-DD')] = 0;
+    dayCounter = addDays(dayCounter, 1);
+  }
 
-  const calendarValues = Object.keys(dateValueMap).map(key => ({
+  for (let timestamp in calendar) {
+    if (calendar.hasOwnProperty(timestamp)) {
+      timestamp = Number(timestamp * 1000) || null;
```

Comment:
Why do not use the old code and change only the last line?

```js
const dateValueMap = Object.keys(calendar)
    .map(ts => Number(ts * 1000) || null)
    .filter(Boolean)
    .reduce((map, current) => {
      const startOfCurrent = format(startOfDay(current), 'YYYY-MM-DD');
      if (startOfCurrent in map) {
        map[startOfCurrent] = map[startOfCurrent] + 1;
      } else {
        map[startOfCurrent] = 1;
      }
      return map;
    }, calendarData);
```

Reply:
I don't know, I liked it better - it's shorter

Reply:
What happens if `timestamp` is null?

Reply:
I guess it all blows up... it really shouldn't ever be null - but yea, you can't count on that, I suppose.
Could add an `if` in there maybe... ?
```
if(timestamp) {
      const startOfTimestampDay = format(startOfDay(timestamp), 'YYYY-MM-DD');
      calendarData[startOfTimestampDay] =
      calendarData[startOfTimestampDay] + 1 || 1;
}
```

Reply:
Turns out they get added to 1970-01-01 (i.e. `null` is treated like 0).  This is out of the range specified by `endOfCalendar` and `startOfCalendar`, so doesn't get displayed.

That said, probably best if they never get added to `calendarData` so adding the `if` seems reasonable to me.

Reply:
I added the condition. Let me know if we want something else.